### ISSUE TITLE
Example SSR with 'use_color_mode' runtime panic

### ISSUE THREAD

**Issue by pintoflager** [Synphonyte/leptos-use#41]
Hi,
thanks for your work on this, crate looks really promising!

I cloned this repo and added browser example of `use_color_mode` and got
```bash
1
2
thread 'main' panicked at /home/user/.cargo/registry/src/index.crates.io-6f17d22bba15001f/js-sys-0.3.64/src/lib.rs:5847:9:
cannot call wasm-bindgen imported functions on non-wasm targets
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /rustc/09df6108c84fdec400043d99d9ee232336fd5a9f/library/std/src/thread/local.rs:246:26:
cannot access a Thread Local Storage value during or after destruction: AccessError
fatal runtime error: thread local panicked on drop
```

Actual code I used for testing:
```rust
  logging::log!("1");
  let color_modes = vec![
      "rust".into(),
      "coal".into(),
      "navy".into(),
      "ayu".into(),
  ];
  
  let options = UseColorModeOptions::default().custom_modes(color_modes.to_owned());
  logging::log!("2");
  let UseColorModeReturn { mode, set_mode, .. } = use_color_mode_with_options(options);
  logging::log!("3");
  // Set theme or use existing
  let current = mode.get_untracked();
  logging::log!("4, active: {:?}", current);

  match current {
      ColorMode::Auto => set_mode.set(ColorMode::Custom("white".into())),
      ColorMode::Light => set_mode.set(ColorMode::Custom("white".into())),
      ColorMode::Dark => set_mode.set(ColorMode::Custom("black".into())),
      ColorMode::Custom(ref s) => {
          if ! color_modes.contains(s) {
              set_mode.set(ColorMode::Custom("black".into()))
          }
      },
  }
  logging::log!("5");
  let UseCycleListReturn { state, next, .. } = use_cycle_list_with_options(
      vec![
          ColorMode::Custom("rust".into()),
          ColorMode::Custom("coal".into()),
          ColorMode::Custom("navy".into()),
          ColorMode::Custom("ayu".into()),
      ],
      UseCycleListOptions::default().initial_value(None)
      // UseCycleListOptions::default().initial_value(Some((mode, set_mode).into())),
  );
  logging::log!("6");
  let on_click = move |_| next();

  view! {
      <button on:click=on_click>{format!("{}", state.get_untracked())}</button>
  }
```

It seems to panic on
```rust
  let UseColorModeReturn { mode, set_mode, .. } = use_color_mode_with_options(options);
```

This should work or did I misunderstood something?

**Comment by maccesch** [comment]
This should work indeed. The error looks like you're using SSR though. Did you enable the `ssr` feature in leptos-use as [described here](https://leptos-use.rs/server_side_rendering.html)?

**Comment by pintoflager** [comment]
It was a unmodified git cloned `ssr` [example](https://github.com/Synphonyte/leptos-use/tree/main/examples/ssr) from this repository which I used as a boilerplate.
And yes, I confirmed this example seems to be configured just like the instructions say.

I only added one component with that test code I posted, no other modifications were made.

**Comment by maccesch** [comment]
I see! Taking a look...

**Comment by maccesch** [comment]
This fix is now released as version 0.8.1
Thanks for reporting this!

**Comment by pintoflager** [comment]
Tested with version 0.8.1 and the same component as the last time -  works a treat!
Thank you!
